src/main/java/org/elasticsearch/client/IndicesAdminClient.java

```diff
@@ -581,6 +581,21 @@
     AnalyzeRequestBuilder prepareAnalyze(@Nullable String index, String text);
 
     /**
+     * Analyze texts under the provided index.
+     *
+     * @param index The index name
+     * @param text  The array of text to analyze
+     */
+    AnalyzeRequestBuilder prepareAnalyze(@Nullable String index, String... text);
```

Comment:
@johtani but in that case users would have to specify the first value via prepareAnalyze and the following ones using addText ?

Along same lines as Lee's comment, how about having a single `AnalyzeRequestBuilder prepareAnalyze(@Nullable String index, String... text);` and drop all the others? It would be a breaking change though for the java api. Otherwise we should have the following without varargs which doesnt look great:
````
AnalyzeRequestBuilder prepareAnalyze(@Nullable String index, String text);
AnalyzeRequestBuilder prepareAnalyze(@Nullable String index, String[] text);
AnalyzeRequestBuilder prepareAnalyze(String text);
AnalyzeRequestBuilder prepareAnalyze(String[] text);
```

I think compromises between the two won't work well.